ucsb-cs156-f24/proj-courses-f24-02
# Added Helpful Error Message When User Adds Two Sections of the Same Course to Schedule

This PR responds with a more helpful error message when a user tries to add two sections of the same course to a personal schedule. Prior to this change, it displayed an error that seems more like a system failure than a user error. Now the error message is more user friendly and understandable for the user.

In order to test this, do the follow:
- Create a personal schedule
- Add a section to the personal schedule
- Try adding another section of the same class to the personal schedule
- You should see an Toast message with the text "Error: You already have a section of this course on your personal schedule; to add this one instead, drop the other one first."

Under the hood, the front end now handles backend error responses more elegantly with more informative Toast messages. This was done by adding a onError handler that was previously missing to sectionTable.js. Additionally, the backend error message in the section controller has been updated to be more detailed

Also, a test case has been added to verify that the onError handler works. Similarly, a backend test has been updated to reflect the more detailed error message.

Dokku deployment: https://courses-dev-leozhu0.dokku-02.cs.ucsb.edu
Fixes #1
